File: CS/TempCollection.Module/Order.cs
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Xpo;
using DevExpress.Xpo.DB;
using System.Collections.Generic;
using System.ComponentModel;

namespace TempCollection.Module {
    [DefaultClassOptions]
    [DefaultProperty("Description")]
    public class Order : BaseObject {
        private string _description;
        public Order(Session session) : base(session) { }
        Dictionary<string, int> dictionaryCore = null;
        public void CollectGroupingInfo() {
            Items.Sorting.Clear();
            Items.Sorting.Add(new SortProperty("Id", SortingDirection.Ascending));
            dictionaryCore = new Dictionary<string, int>();
            foreach (Item item in Items) {
                if (dictionaryCore.ContainsKey(item.Id)) {
                    dictionaryCore[item.Id] += item.Quantity;
                }
                else {
                    dictionaryCore.Add(item.Id, item.Quantity);
                }
            }
        }
        public void UpdateGrouping() {
            CollectGroupingInfo();
            Session.Delete(Items);
            foreach (KeyValuePair<string, int> pair in dictionaryCore) {
                Item newItem = new Item(Session);
                newItem.Id=pair.Key;
                newItem.Quantity = pair.Value;
                Items.Add(newItem);
            }
        }
        public string Description {
            get { return _description; }
            set { SetPropertyValue("Description", ref _description, value); }
        }
        [Association("RealAssociation", typeof(Item)), Aggregated]
        public XPCollection<Item> Items {
            get { return GetCollection<Item>("Items"); }
        }
    }
}

File: VB/TempCollection.Module/Order.vb
Imports Microsoft.VisualBasic
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Xpo
Imports DevExpress.Xpo.DB
Imports System.Collections.Generic
Imports System.ComponentModel

Namespace TempCollection.Module
	<DefaultClassOptions, DefaultProperty("Description")> _
	Public Class Order
		Inherits BaseObject
		Private _description As String
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Private dictionaryCore As Dictionary(Of String, Integer) = Nothing
		Public Sub CollectGroupingInfo()
			Items.Sorting.Clear()
			Items.Sorting.Add(New SortProperty("Id", SortingDirection.Ascending))
			dictionaryCore = New Dictionary(Of String, Integer)()
			For Each item As Item In Items
				If dictionaryCore.ContainsKey(item.Id) Then
					dictionaryCore(item.Id) += item.Quantity
				Else
					dictionaryCore.Add(item.Id, item.Quantity)
				End If
			Next item
		End Sub
		Public Sub UpdateGrouping()
			CollectGroupingInfo()
			Session.Delete(Items)
			For Each pair As KeyValuePair(Of String, Integer) In dictionaryCore
				Dim newItem As New Item(Session)
				newItem.Id=pair.Key
				newItem.Quantity = pair.Value
				Items.Add(newItem)
			Next pair
		End Sub
		Public Property Description() As String
			Get
				Return _description
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Description", _description, value)
			End Set
		End Property
		<Association("RealAssociation", GetType(Item)), Aggregated> _
		Public ReadOnly Property Items() As XPCollection(Of Item)
			Get
				Return GetCollection(Of Item)("Items")
			End Get
		End Property
	End Class
End Namespace